Lib/test/test_generators.py

```diff
@@ -268,6 +268,105 @@ def loop():
         #This should not raise
         loop()
 
+
+class ModifyTest(unittest.TestCase):
+    sequences = [
+        range(0),
+        range(20),
+        [1, 2, 3],
+        (2,),
+        {13, 48, 211},
+        frozenset((15, 8, 6)),
+        {1: 2, 3: 4},
+    ]
+
+    non_sequences = [
+        None,
+        42,
+        3.0,
+        2j,
+    ]
+
+    def modify_f_locals(self, g, local, new_iter):
+        g.gi_frame.f_locals[local] = new_iter
+        return g
+
+    def new_gen_from_gi_code(self, g, new_iter):
+        generator_func = types.FunctionType(g.gi_code, {})
+        return generator_func(new_iter)
+
+    def test_modify_genexpr(self):
+        def genexpr():
+            return (x for x in range(10))
+
+        def get_genexpr_f_locals(self, new_iter):
+            return self.modify_f_locals(genexpr(), '.0', new_iter)
+
+        def get_genexpr_gi_code(self, new_iter):
+            return self.new_gen_from_gi_code(genexpr(), new_iter)
+
+        err_msg_pattern_genexpr = "'%s' object is not an iterator"
```

Comment:
I also feel like this part in test can be simplified: we can just keep the "object is not an iterator" part without the type formating, which is not essential to this test in my opinion.

But, others might feel otherwise.